[excel] (Custom Functions) Refresh low engagement custom functions articles (#5848)
OfficeDev/office-js-docs-pr: docs/excel/custom-functions-volatile.md

---
ms.date: 07/28/2026
ms.topic: concept-article
description: Create an Excel custom function that recalculates whenever Excel recalculates, even when its arguments haven't changed.
title: Volatile values in functions
ms.localizationpriority: medium
ai-usage: ai-assisted
---

# Volatile values in functions

> A volatile function recalculates whenever Excel recalculates, even if none of the function's arguments changed. Use a volatile custom function for values that can change independently of their inputs, such as dates, times, random numbers, and simulation results.

Reviewer: For global English, we try to avoid possessive 's (it's hard to translate). Do you think this is still accurate? 
```suggestion
A volatile function recalculates whenever Excel recalculates, even if none of the function arguments changed. Use a volatile custom function for values that can change independently of their inputs, such as dates, times, random numbers, and simulation results.
```

Author: Yeah, I don't think this reads any differently the way you wrote. Thanks!

> [!NOTE]
> Several built-in Excel functions are volatile, such as `RAND`, `TODAY`, and `NOW`.

> To mark an autogenerated custom function as volatile, add the `@volatile` tag to its JSDoc comment. For more information about autogenerated metadata, see [Autogenerate JSON metadata for custom functions](custom-functions-json-autogeneration.md).

Reviewer: I don't think customers consider custom functions with autogenerated JSON metadata "autogenerated custom functions". We could say this, since autogenerating is the recommended approach. 
```suggestion
To mark a custom function as volatile, add the `@volatile` tag to its JSDoc comment. For more information about custom functions JSON metadata, see [Autogenerate JSON metadata for custom functions](custom-functions-json-autogeneration.md).
```

## Create a volatile custom function

The following example uses `@volatile` to create a custom function that simulates rolling a six-sided die.

> :::image type="content" source="../images/six-sided-die.gif" alt-text="GIF showing a custom function returning a random value to simulate rolling a six-sided die.":::

Reviewer: Technically accurate but really has me laughing.

Author: Maybe it would have lived if it had more sides...
